src/System.Text.JsonLab/System/Text/Json/JsonThrowHelper.cs

```diff
@@ -19,6 +19,17 @@ private static ArgumentException GetArgumentException(string message)
             return new ArgumentException(message);
         }
 
+        public static void ThrowArgumentExceptionInvalidUtf8String()
```

Comment:
Why is this beneficial? Isn't `ThrowArgumentException(string message)` good enough?

Reply:
The il and asm are larger as it has to load the string reference before calling the Throw helper

Via String
```asm
G_M53912_IG05:
       48B9683016EFC1010000 mov      rcx, 0x1C1EF163068
       488B09               mov      rcx, gword ptr [rcx]
       E81FFCFFFF           call     JsonThrowHelper:ThrowArgumentException(ref)
       CC                   int3   
```
Via Enum is smaller as its just passing a numeric const
```asm
G_M12337_IG05:
       33C9                 xor      ecx, ecx
       E8F2FBFFFF           call     JsonThrowHelper:ThrowArgumentException(int)
       CC                   int3     
```
No Params is just a call
```asm
G_M36770_IG05:
       E8D4FBFFFF           call     JsonThrowHelper:ThrowArgumentExceptionInvalidUtf8String()
       CC                   int3   
```

Reply:
e.g. if you are inlining it all over the place, or duplicating it due to generics, better to be smaller